pkg/controller/nodeipam/ipam/controller.go

```diff
@@ -92,6 +92,10 @@ func NewController(
 		return nil, err
 	}
 
+	if _, err := c.set.ProbeCIDRLeft(); err != nil {
```

Comment:
why not

```golang
cidr, err := c.set.AllocateNext()
switch err {
  case ErrCIDRRangeNoCIDRsRemaining:
    return nil, fmt.Errorf("no free IP space for pods in the CIDR range")
  case nil:
    c.set.Release(cidr)
    return c, nil
  default:
    return nil, fmt.Error("error when testing for free CIDR range: %v", err)
}
```

Reply:
You don't have to add all of the other code.

Reply:
Done. In my perspective, check there is a space is more reasonable than allocate and release a cidr.